docs(accessibility-fundamentals): Updates content as part of audit. (#3762)
patternfly/patternfly-org: packages/documentation-site/patternfly-docs/content/developer-resources/accessibility-fundamentals.md

---
id: About
section: accessibility
---

# Accessibility fundamentals 

Accessibility is a foundation of PatternFly, so if you contribute to PatternFly or use it for your products you should first familiarize yourself with our accessibility guidelines and recommendations.

> No user should feel left behind when using your product. The goal of accessible design is to remove barriers and create inclusive product experiences that work for everyone, regardless of ability. This is best achieved when accessibility is considered early in the design and development process. 

Reviewer: Not a blocker: what do you think about adding onto the last sentence, something like, "This is best achieved when accessibility is considered early in the design and development process, but it's never too late to start considering it."?

There’s always room to improve accessibility. That’s why our guidelines will continually evolve. Your feedback can help support PatternFly's accessibility, so we welcome you to contribute to our documentation or [get involved in our GitHub discussions](https://github.com/orgs/patternfly/discussions). 

## Assistive technologies

> A core consideration for accessibility practices is assistive technologies that people may use to access your product. 

Reviewer: I'm on the fence of adding some verbiage somewhere in this section that basically lets readers know that there *are* other assistive tech, without making them feel like they need to consider everything all at once (and thus overwhelming themselves). @jessiehuff wdyt?

### Keyboard

> Some users can’t use a mouse and rely on other methods of navigation instead, like a keyboard. A keyboard allows users to navigate through UI elements in the order that they appear on the page, typically using the tab key. Keyboard users should be able to do anything that mouse users can, including viewing text that is shown upon hover or in popups.  

Reviewer: ```suggestion
Some users can’t use a mouse and rely on other methods of navigation instead, like a keyboard. A keyboard allows users to navigate through UI elements in the order that they appear on the page, typically using the <kbd>Tab</kbd> key. Keyboard users should be able to do anything that mouse users can, including viewing text that is shown upon hover or in popups.  
```

### Screen readers
When a user has impaired vision (temporary or permanent) that prevents them from viewing a screen, they can use a screen reader that reads visible and non-visible content aloud. Non-visible content includes hidden text, like tooltips, and alternative text (alt text) for icons, images, and links. 

## Understanding users’ needs
Great user experiences don’t just happen. They’re designed, tested, and refined with users in mind. To develop inclusive products, it’s important to understand the varying needs of all users and consider the assistive tools and methods that they use.

This section provides information to help you better understand and address the needs of users [with diverse abilities](https://www.a11yproject.com/posts/accessibility-is-blind-people/). It is important to note that some users might fall into multiple groups, and some might use tools created for a different group.

### Motor control

> Users with reduced motor control often use a keyboard or computer mouse to access content.

Reviewer: Not 100% about the wording, but what do you think about an addition similar to the following? It kind of ties into the bullet about target area size.

```suggestion
Users with reduced motor control often use a keyboard or computer mouse to access content. Some users may also have difficulty keeping a mouse steady when trying to interact with content.
```

> To support users with reduced motor control, remember that: 

Author: ```suggestion
To design for users with reduced motor control, remember that: 
```

- Users who rely on a keyboard need elements that are keyboard accessible and highly visible when in focus.

> - Users who rely on a mouse or touch need target areas that are large enough to be hit easily.

Reviewer: Instead of "hit", maybe "activate" or "click"?

> - Users who rely on a mouse or touch need target areas that are large enough to be hit easily.

Author: ```suggestion
- Users who rely on a mouse or touch need target areas that are large enough to be easily clicked and selected.
```

### Vision

#### No vision 
Users with no vision rely on screen readers to access web sites and applications. These users often navigate a page by observing specific elements, like headers, links, or form elements.

> To design for users with no vision:

Reviewer: The previous section uses "To support users..." verbiage, vs "To design for..." here and the other sections. Would we want to keep it consistent and use one prefacing text?

- Use [semantic elements](w3schools.com/html/html5_semantic_elements.asp).
- Write labels that are meaningful when pulled out of context.

#### Low vision
Users with low vision can have different needs depending on the nature of their visual impairment, such as difficulty with color differentiation, blurriness, or lack of vision in central or peripheral areas.

To design for users with low vision:

- Interfaces should not rely on color alone to communicate information.
- Palettes must have sufficient contrast.
- Layouts should be responsive to font size increases.

### Cognitive processing 
Users who have difficulty processing information benefit from well-written content.

To design for users with cognitive processing issues:

- Write content that is clear, concise, and easy to scan.
- Consider visual hierarchy and chunk content into short, related sections. 
- Avoid long paragraphs.

### Accessibility spectrum

> While there are a few main accessibility needs, it is not an exhaustive list. Often the “Persona Spectrum”, coined by [Microsoft](https://download.microsoft.com/download/b/0/d/b0d4bf87-09ce-4417-8f28-d60703d672ed/inclusive_toolkit_manual_final.pdf), can be used to understand related mismatches and motivations across a spectrum of permanent, temporary, and situational scenarios.

Author: agree - https://inclusive.microsoft.design/ it's linked on this page, but there are no heading links, so it seems hard to direct users to the right link on the page. Adding that PDF verbiage sounds like a good solution though!

> While there are a few main accessibility needs, it is not an exhaustive list. Often the “Persona Spectrum”, coined by [Microsoft](https://download.microsoft.com/download/b/0/d/b0d4bf87-09ce-4417-8f28-d60703d672ed/inclusive_toolkit_manual_final.pdf), can be used to understand related mismatches and motivations across a spectrum of permanent, temporary, and situational scenarios.

Author: ```suggestion
While there are a few main accessibility needs, it is not an exhaustive list. Often the “Persona Spectrum”, coined by Microsoft [(Download the Inclusive Guidebook as a PDF)](https://download.microsoft.com/download/b/0/d/b0d4bf87-09ce-4417-8f28-d60703d672ed/inclusive_toolkit_manual_final.pdf), can be used to understand related mismatches and motivations across a spectrum of permanent, temporary, and situational scenarios.
```

While accessibility tends to focus on those with permanent disabilities, everyone benefits from accessible products:

- Accessible content supports better comprehension across language fluency and literacy levels.
- Alternative text (alt text) makes images accessible to users with low bandwidth connections or older technologies that can’t load the images.
- Closed captioning benefits those in crowded areas, those learning how to read, and those learning a new language.

Accessibility takes **all users** into account.

> <p className="ws-p pf-v5-u-screen-reader">
>   The following image shows how accessibility is a spectrum. Different kinds of cognitive and physical impairments impact how users interact with products. All levels of impairment should be considered when building accessible products.
> </p>

Author: ```suggestion
```

> <img src="./PF-a11y.png" alt="Accessibility spectrum image" />

Reviewer: Wondering if we could remove the hidden paragraph and just update the image alt text...

`alt="An accessibility spectrum showing examples of the situational, temporary, and permanent situations for different impairments."` maybe?

Was removing this image also something we'd want to consider in a follow up? I think we discussed the image in general a bit in the google doc, but not sure if we landed 100% on anything.

Author: I personally think the image should be reworked at the very least, but also wasn't sure where we landed. It's tough because I do like the information it conveys, it's just so large. 

Maybe it should just be a followup to remove/replace/repurpose the image? _An idea I had was to cut the image up so that each "row" is placed within the relevant section (for example, the "sight" row is placed under the "vision" section), but not everything in that image has a place in that case._

Reviewer: That could definitely be a followup. I do like at least cutting the image up to several, smaller ones as well.

> <img src="./PF-a11y.png" alt="Accessibility spectrum image" />

Author: ```suggestion
<img src="./PF-a11y.png" alt="An accessibility spectrum showing examples of the situational, temporary, and permanent situations for different impairments." />
```

---

> id: Accessibility fundamentals

Author: ```suggestion
id: About
```

section: accessibility
---

> 

Author: ```suggestion

# Accessibility fundamentals 

```

</p>

<img src="./PF-a11y.png" alt="Accessibility spectrum image" />
<br />
<br />

## Experience parity
We believe that all abilities should be treated equally. There should be parity in the experience of all users—one user group shouldn't be prioritized over the other.
To help you achieve this, consider these guidelines:

1. Content should be optimized for all input types: touch, mouse, and keyboard.
  - Don’t optimize the experience for one input type at the expense of another.
  - Contents that are mouse-accessible should also be accessible using touch or a keyboard.
  - Don’t show interactive elements on hover. Interactive elements that can display in a pop-up must display on click, touch, or enter-key events.

2. Screen reader content should match visibly rendered content (refer to the [first note for `aria-hidden` state](https://www.w3.org/TR/wai-aria/#aria-hidden)).

3. There should be parity between hover and focus events. Any information that’s available on hover for the mouse user should be available on keyboard focus.

>   - Make content should available to a screen reader by using `aria-describedby` (refer to [Tooltips & Toggletips example from Inclusive Components](https://inclusive-components.design/tooltips-toggletips/)).

Reviewer 2: In general, this all looks really good! :) There's a typo in this sentence that really stood out to me though. Since we're separating that bullet from the guideline, can we add a little bit of specificity? Maybe something like, "Make content appearing on hover available to a screen reader by using `aria-describedby`."

>   - Make content should available to a screen reader by using `aria-describedby` (refer to [Tooltips & Toggletips example from Inclusive Components](https://inclusive-components.design/tooltips-toggletips/)).

Author: ```suggestion
  - Make content that appears on hover available to a screen reader by using `aria-describedby` (refer to [Tooltips & Toggletips example from Inclusive Components](https://inclusive-components.design/tooltips-toggletips/)).
```

---
id: About
section: accessibility
---

# Accessibility fundamentals 

Accessibility is a foundation of PatternFly, so if you contribute to PatternFly or use it for your products you should first familiarize yourself with our accessibility guidelines and recommendations.

No user should feel left behind when using your product. The goal of accessible design is to remove barriers and create inclusive product experiences that work for everyone, regardless of ability. This is best achieved when accessibility is considered early in the design and development process, but it's never too late to start prioritizing accessibility.

There’s always room to improve accessibility. That’s why our guidelines will continually evolve. Your feedback can help support PatternFly's accessibility, so we welcome you to contribute to our documentation or [get involved in our GitHub discussions](https://github.com/orgs/patternfly/discussions). 

## Assistive technologies

> A core consideration for accessibility practices is assistive technologies that people may use to access your product. 

Author: ```suggestion
A core consideration for accessibility practices is the assistive technologies that people may use to access your product. While there are many kinds of assistive technologies, PatternFly focuses on accommodating keyboard and screen reader users, as these are the most common.
```
@thatblindgeye what about something like this?

Reviewer: I think that works! Maybe dropping the last bit so it's just "While there are many kinds of assistive technologies, PatternFly focuses primarily on keyboard and screen readers." (or "...accommodating users of keyboards and screen readers.", just for a "user first" approach).
